clang/tools/archgen-mlir/ArchGenMLIR.cpp

```diff
@@ -0,0 +1,927 @@
+//===- ArchGenMLIR.cpp ---------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Hybrid(LLVMIR, MLIR) IR-gen used for the compiler implementation based of
+// archgenlib
```

Comment:
```suggestion
// Hybrid (LLVMIR, MLIR) IR-gen used for the compiler implementation based on
// archgenlib
```
This is a depressingly short description. :-) No inspiration?

Reply:
none what so ever.